### ISSUE TITLE
关于Region Decomposition模块的咨询

### ISSUE THREAD

**Issue by YfWangZJU** [qcraftai/distill-bev#7]
你好！
bevformer_distill.py中有
```
        if self.distill_params['fp_as_foreground'][index] != 'none' and self._epoch >= self.distill_params['fp_epoch']:
            fp_mask, fp_scale_mask, fp_points_number = self.add_fp_as_fg_bbox(student_H, student_W,
                self.distill_params['fp_as_foreground'][index], foreground_mask,
                              teacher_preds, gt_bboxes_3d)
            bg_mask[fp_mask != 0] = 0
            bg_points_number = student_H * student_W - torch.sum(foreground_mask, dim=(1,2,3))
            for b in range(B):
                # in case an extremely weak model predicts all the points as positive
                if bg_points_number[b] > fp_points_number[b]:
                    bg_scale_mask[b][:] = 1.0 / (bg_points_number[b] - fp_points_number[b])
                else:
                    bg_scale_mask[b][:] = 0
```

config文件中又设置了
```
fp_as_foreground='none',
fp_weight=0,
```
是否意味着开源代码中region decomposition mask M在此设置下其实并未生效？config文件中的fp_weight是否就是论文公式1中的η？论文补充材料中写道

> In all experiments, we set the region decomposition and
> spatial attention related hyper-parameters as η = 20, τ = 0.5,
> γ = 0.1.

此处的η如果是公式1中的η，20的设置似乎会使得kd_fp_bg_feat_loss_head_head权重过高？

**Comment by dingwenli** [comment]
你好，请参照scripts里的参数设置，例如scripts/teacher_to_bevdepth4d/centerpoint2bevdepth.sh. 在代码中fp_weight = \$\eta\$ * fg_feat_loss_weight，也就是20*fg_feat_loss_weight，不是直接设为20

**Comment by YfWangZJU** [comment]
原来如此，感谢解答！不知我这么理解是否正确：针对不同的教师+学生模型组合，region decomposition mask需要酌情使用？我观察到

> scripts/teacher_to_bevformer/exp_lidar_r50.sh + 
configs/lidar2camera_bev_distillation/teacher_to_bevformer/lidarformer_to_bevformer_nus_1x1conv_r50.py

的fp_weight为0，mask M模块看起来被屏蔽了，而在你举例的

> scripts/teacher_to_bevdepth4d/centerpoint2bevdepth.sh + 
configs/lidar2camera_bev_distillation/centerpoint_pillar_to_bevdepth4d_r50/centerpoint_02pillar_second_secfpn_circlenms_8x4_cyclic_20e_nus_to_bevdepth4d_r50.py

中fp_weight为6e-2，region decomposition mask生效了。

**Comment by dingwenli** [comment]
是的
